### ISSUE TITLE
Blog post subscription notification emails lacks In-Reply-To header field

### ISSUE THREAD

**Issue by mmd-osm** [openstreetmap/openstreetmap-website#2430]
I'm subscribed to a blog post and receive lots of emails by users replying to that post. As those emails don't include an `In-Reply-To` header, my inbox fills up with pages and pages of messages, rather than producing a threaded view of all emails. Not sure if Rails suppors this at all, but it would be quite handy.

**Comment by tomhughes** [comment]
Well what value would you give it? I'm not even sure what generates the message IDs, ie whether rails does it or whether it leaves it for exim to do, but I am sure it doesn't record them anywhere so it has no idea what value to use in the future.

There's also no way to know which comment a future comment refers to, so would you just have all comments as replies to the original entry? But that never has a message ID at all because no email was ever sent for it... Even if we invented one I'm not sure what a mail client would do with a bunch of messages that claim to reply to a message it has never seen?

**Comment by mmd-osm** [comment]
As an example, GitHub uses an `In-Reply-To: <openstreetmap/openstreetmap-website/pull/2229@github.com>` style header for replies. IIRC, this also works when you hit the "Subscribe" button where you never received the original message for an issue/pull request.

We could use a similar fake header, like `In-Reply-To: <openstreetmap-diary-123456@openstreetmap.org>`. We know that the message should be created for a diary comment, and each diary has a unique numeric identifier.

**Comment by mmd-osm** [comment]
First idea for an implementation would be something along those lines:

```ruby
diff --git a/app/mailers/notifier.rb b/app/mailers/notifier.rb
index b12599981..98d4946d8 100644
--- a/app/mailers/notifier.rb
+++ b/app/mailers/notifier.rb
@@ -99,6 +99,8 @@ class Notifier < ApplicationMailer
 
       attach_user_avatar(comment.user)
 
+      headers['In-Reply-To'] = "osm-diary-#{comment.diary_entry.id}@#{Settings.server_url}"
+
       mail :from => from_address(comment.user.display_name, "c", comment.id, comment.digest, recipient.id),
            :to => recipient.email,
            :subject => I18n.t("notifier.diary_comment_notification.subject", :user => comment.user.display_name)
@@ -136,6 +138,8 @@ class Notifier < ApplicationMailer
       @author = @commenter
       attach_user_avatar(comment.author)
 
+      headers['In-Reply-To'] = "osm-note-#{comment.note.id}@#{Settings.server_url}"
+
       subject = if @owner
                   I18n.t("notifier.note_comment_notification.#{@event}.subject_own", :commenter => @commenter)
                 else
@@ -166,6 +170,8 @@ class Notifier < ApplicationMailer
 
       attach_user_avatar(comment.author)
 
+      headers['In-Reply-To'] = "osm-changeset-#{comment.changeset.id}@#{Settings.server_url}"
+
       mail :to => recipient.email, :subject => subject
     end
   end

```

**Comment by gravitystorm** [comment]
> First idea for an implementation would be something along those lines:

Please could you post this code as a PR or as a link to a commit. Posting code into a comment means everyone else has to copy+paste+commit etc. PRs are even better since that runs CI, allows reviewers to expand the diff to adjacent lines etc.

**Comment by mmd-osm** [comment]
Sure, that makes sense. PR is available in #2462 now.
